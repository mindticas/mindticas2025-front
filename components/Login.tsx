import {
    Box,
    Button,
    Field,
    Heading,
    Input,
    Stack,
    VStack,
    InputAddon,
} from '@chakra-ui/react';
import { Eye, EyeOff } from 'lucide-react';
import React, { useState } from 'react';
import { InputGroup } from './ui/input-group';

export default function Login() {
    const [showPassword, setShowPassword] = useState(false);

    const togglePasswordVisibility = () => {
        setShowPassword(!showPassword);
    };

    return (
        <Box
            w='100vw'
            h='100vh'
            bg='white'
            display='flex'
            justifyContent='center'
            alignItems='center'
            p={4}
        >
            <Box
                bg='white'
                p={[4, 8]}
                borderRadius='md'
                boxShadow='lg'
                w={['90%', '100%', '400px']}
            >
                <form>
                    <VStack gap='6'>
                        <Heading as='h1' size='xl' textAlign='center' mb={5}>
                            Iniciar sesión
                        </Heading>
                        <Stack gap='6' w='full'>
                            <Field.Root>
                                <Field.Label fontSize='md' mb='1'>
                                    Usuario
                                </Field.Label>
                                <Input
                                    name='username'
                                    px={4}
                                    py={6}
                                    size='md'
                                    placeholder='Ingresa tu usuario'
                                />
                            </Field.Root>
                            <Field.Root>
                                <Field.Label fontSize='md' mb='1'>
                                    Contraseña
                                </Field.Label>
                                <InputGroup w='full' position='relative'>
                                    <>
                                        <Input
                                            name='password'
                                            type={
                                                showPassword
                                                    ? 'text'
                                                    : 'password'
                                            }
                                            px={4}
                                            py={6}
                                            size='md'
                                            placeholder='Ingresa tu contraseña'
                                            pr='40px'
                                        />
                                        <InputAddon
                                            position='absolute'
                                            right='10px'
                                            top='50%'
                                            transform='translateY(-50%)'
                                            zIndex={2}
                                            background='transparent'
                                            border='none'
                                            cursor='pointer'
                                            onClick={togglePasswordVisibility}
                                        >
                                            {showPassword ? (
                                                <EyeOff size={18} />
                                            ) : (
                                                <Eye size={18} />
                                            )}
                                        </InputAddon>
                                    </>
                                </InputGroup>
                            </Field.Root>
                            <Button
                                type='submit'
                                bg='black'
                                color='white'
                                _hover={{ bg: 'gray.800' }}
                                w='full'
                            >
                                Enviar
                            </Button>
                        </Stack>
                    </VStack>
                </form>
            </Box>
        </Box>
    );
}
