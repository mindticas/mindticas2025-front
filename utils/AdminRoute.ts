import Layout from '@/components/Layout';
import { Box } from '@chakra-ui/react';

export class AdminRoute {
    // Fixed route for admin
    private static route: string = '/admin';

    // Returns the Layout component if the pathname contains '/admin', otherwise returns box
    static getLayout(pathname: string) {
        return pathname.includes(this.route) ? Layout : Box;
    }
}
