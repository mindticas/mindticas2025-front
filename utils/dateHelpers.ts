// Gets the name of the day according to the full calendar format
export const getDayName = (dateStr: string): string => {
    const [year, month, day] = dateStr.split('-').map(Number)
    const date = new Date(year, month -1, day)
    return date.toLocaleDateString('es-ES', { weekday: 'long'})
}
// Get the milliseconds in a day
export function addDays(date: Date, days: number): Date {
    const milisecondsPerDay = 24 * 60 * 60 * 1000; 
    return new Date(date.getTime() + days * milisecondsPerDay)
}