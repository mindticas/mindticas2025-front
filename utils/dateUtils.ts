import { DateTime } from "luxon";

interface DateTimeParts {
    date: string;
    time: string;
}

// * @returns Object with (YYYY-MM-DD) y time (HH:MM)

export const splitDateTimeFromISO = (
    isoString: string | null | undefined,
): DateTimeParts => {
    if (!isoString) return { date: '', time: '' };

    const dateObj = new Date(isoString);

    if (isNaN(dateObj.getTime())) {
        return { date: '', time: '' };
    }

    // format date (YYYY-MM-DD)
    const date = DateTime.fromJSDate(dateObj)
       .setZone('local')
        .toFormat('yyyy-MM-dd');

    // format time (HH:MM)
    const time = dateObj
        .toLocaleTimeString('es-ES', {
            hour: '2-digit',
            minute: '2-digit',
            hour12: false,
        })
        .replace(/\./g, '');

    return { date, time };
};
