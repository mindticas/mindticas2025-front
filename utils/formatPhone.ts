export default function formatPhone(phone: string): string {
    // Remove everything that is not digits
    const digits = phone.replace(/\D/g, '');
    // Applies format only if it has 10 digits
    if (digits.length === 10) {
        return `${digits.slice(0, 3)}-${digits.slice(3, 6)}-${digits.slice(6)}`;
    }
    return phone; // Returns the original if it doesn't match
}
