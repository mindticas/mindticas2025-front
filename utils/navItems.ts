import { BarChart, Calendar, Scissors, UserCog, Users } from 'lucide-react';

export const navItems = [
    { name: 'Citas', href: '/admin/', icon: Calendar },
    { name: 'Clientes', href: '/admin/clients', icon: Users },
    { name: 'Empleados', href: '/admin/employees', icon: UserCog },
    { name: 'Tratamientos', href: '/admin/treatments', icon: Scissors },
    { name: 'Reportes', href: '/admin/reports', icon: BarChart },
];
