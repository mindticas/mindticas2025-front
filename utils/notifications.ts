import { toaster } from '@/components/ui/toaster';

type ToastType = 'success' | 'error' | 'info' | 'warning';

interface ToastOption {
    duration?: number;
    title: string;
    description?: string;
}

const showToast = (type: ToastType, options: ToastOption) => {
    toaster.create({
        type,
        duration: options.duration || 5000,
        title: options.title,
        description: options.description,
    });
};

const TreatmentNotification = {
    createSuccess: () =>
        showToast('success', { title: 'Tratamiento creado con éxito' }),
    updateSuccess: () =>
        showToast('success', { title: 'Tratamiento actualizado con éxito' }),
    deleteSuccess: () =>
        showToast('success', { title: 'Tratamiento eliminado con éxito' }),
    createError: (message?: string) =>
        showToast('error', {
            title: message || 'Error al crear el tratamiento',
        }),
    updateError: (message?: string) =>
        showToast('error', {
            title: message || 'Error al actualizar el tratamiento0',
        }),
    deleteError: () =>
        showToast('error', { title: 'Error al eliminar el tratamiento' }),
    noChanges: () =>
        showToast('info', {
            title: 'No se detectaron cambios para actualizar',
        }),
};

export default TreatmentNotification;
