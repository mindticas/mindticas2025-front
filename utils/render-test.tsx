import { Provider } from '@/components/ui/provider';
import { BookingProvider } from '@/context/BookingContext';
import { render as rtlRender } from '@testing-library/react';

export function render(ui: React.ReactNode) {
    return rtlRender(<>{ui}</>, {
        wrapper: (props: React.PropsWithChildren) => (
            <Provider>
                <BookingProvider>{props.children}</BookingProvider>
            </Provider>
        ),
    });
}
