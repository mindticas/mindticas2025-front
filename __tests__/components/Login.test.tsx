import Login from '@/components/Login';
import { render } from '@/utils/render-test';

describe('Login Snapshot', () => {
    it('should match the snapshot', () => {
        const { asFragment } = render(<Login />);
        expect(asFragment()).toMatchSnapshot();
    });
});
