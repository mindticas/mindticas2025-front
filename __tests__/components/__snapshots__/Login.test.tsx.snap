// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Snapshot should match the snapshot 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    ((e,s,n,m,o,l,u,h)=&gt;{let a=document.documentElement,v=["light","dark"];function p(c){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&l?o.map(f=&gt;l[f]||f):o;k?(a.classList.remove(...S),a.classList.add(c)):a.setAttribute(y,c)}),C(c)}function C(c){h&&v.includes(c)&&(a.style.colorScheme=c)}function d(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(m)p(m);else try{let c=localStorage.getItem(s)||n,y=u&&c==="system"?d():c;p(y)}catch(c){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div
    class="css-1p1q308"
  >
    <div
      class="css-1234vry"
    >
      <form>
        <div
          class="chakra-stack css-11vffse"
        >
          <h1
            class="chakra-heading css-1dtsck9"
          >
            Iniciar sesión
          </h1>
          <div
            class="chakra-stack css-fn1s88"
          >
            <div
              class="chakra-field__root css-13f87cb"
              data-part="root"
              data-scope="field"
              id="field:::r0:"
              role="group"
            >
              <label
                class="chakra-field__label css-9de5sl"
                data-part="label"
                data-scope="field"
                for=":r0:"
                id="field:::r0:::label"
              >
                Usuario
              </label>
              <input
                class="chakra-input css-1hb1osm"
                data-part="input"
                data-scope="field"
                id=":r0:"
                name="username"
                placeholder="Ingresa tu usuario"
              />
            </div>
            <div
              class="chakra-field__root css-13f87cb"
              data-part="root"
              data-scope="field"
              id="field:::r1:"
              role="group"
            >
              <label
                class="chakra-field__label css-9de5sl"
                data-part="label"
                data-scope="field"
                for=":r1:"
                id="field:::r1:::label"
              >
                Contraseña
              </label>
              <div
                class="css-r4izcz"
              >
                <input
                  class="chakra-input css-1gk9sfn"
                  data-part="input"
                  data-scope="field"
                  id=":r1:"
                  name="password"
                  placeholder="Ingresa tu contraseña"
                  type="password"
                />
                <div
                  class="css-ds5jk8"
                >
                  <svg
                    class="lucide lucide-eye"
                    fill="none"
                    height="18"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="18"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M2.062 12.348a1 1 0 0 1 0-.696 10.75 10.75 0 0 1 19.876 0 1 1 0 0 1 0 .696 10.75 10.75 0 0 1-19.876 0"
                    />
                    <circle
                      cx="12"
                      cy="12"
                      r="3"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <button
              class="chakra-button css-723201"
              type="submit"
            >
              Enviar
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
